chips/nrf52/src/uart.rs

```diff
@@ -448,6 +440,14 @@ impl<'a> Uarte<'a> {
         self.set_tx_dma_pointer_to_buffer();
 
         let regs = &*self.registers;
+
+        // Make sure we clear the endtx interrupt since that is what we rely on
+        // to know when the DMA TX finishes. Normally, we clear this interrupt
+        // as we handle it, so this is not necessary. However, a bootloader (or
+        // some other startup code) may have setup TX interrupts, and there may
+        // be one pending. We clear it to be safe.
+        regs.event_endtx.write(Event::READY::CLEAR);
```

Comment:
Do we need to do this every time when setting up the buffer, or is once in initialize enough?

Reply:
Turns out we don't. Fixed.